tests/jax/components/building/datasets_test.py

```diff
@@ -0,0 +1,145 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for TransitionDataset and TrajectoryDataset classes for Jax-based Mava systems"""
+
+from types import SimpleNamespace
+from typing import Any, Dict
+
+import pytest
+import reverb
+from tensorflow.python.framework import dtypes, ops
+
+from mava import specs
+from mava.adders import reverb as reverb_adders
+from mava.components.jax.building.datasets import TrajectoryDataset, TransitionDataset
+from mava.systems.jax.builder import Builder
+from tests.jax.mocks import make_fake_env_specs
+
+env_spec = make_fake_env_specs()
+
+
+def adder_signature_fn(
+    ma_environment_spec: specs.MAEnvironmentSpec,
+    extras_specs: Dict[str, Any],
+) -> Any:
+    """signature function that helps in building a simple server"""
+    return reverb_adders.ParallelNStepTransitionAdder.signature(
+        ma_environment_spec=ma_environment_spec, extras_specs=extras_specs
+    )
+
+
+class MockBuilder(Builder):
+    """Mock Builder component"""
+
+    def __init__(self) -> None:
+        self.simple_server = reverb.Server(
+            [
+                reverb.Table.queue(
+                    name="table_0",
+                    max_size=100,
+                    signature=adder_signature_fn(env_spec, {}),
+                )
+            ]
+        )
+        data_server_client = SimpleNamespace(
+            server_address=f"localhost:{self.simple_server.port}"
+        )
+        trainer_id = "table_0"
+        self.store = SimpleNamespace(
+            data_server_client=data_server_client, trainer_id=trainer_id
+        )
+
+
+@pytest.fixture
+def mock_builder() -> MockBuilder:
+    """Create builder mock"""
+    return MockBuilder()
+
+
+def test_init_transition_dataset() -> None:
+    """Test initiator of TransitionDataset component"""
+    transition_dataset = TransitionDataset()
+
+    assert transition_dataset.config.sample_batch_size == 256
+    assert transition_dataset.config.prefetch_size == None
+    assert transition_dataset.config.num_parallel_calls == 12
+    assert transition_dataset.config.max_in_flight_samples_per_worker == None
+    assert transition_dataset.config.postprocess == None
+
+
+def test_on_building_trainer_dataset_transition_dataset(
+    mock_builder: MockBuilder,
+) -> None:
+    """Test on_building_trainer_dataset of TransitionDataset Component
+
+    Args:
+        mock_builder: Builder
+    """
+    transition_dataset = TransitionDataset()
+    transition_dataset.on_building_trainer_dataset(builder=mock_builder)
+
+    dataset = mock_builder.store.dataset._dataset._map_func._func(1)._dataset
```

Comment:
Please add a comment here to explain how this gives us the dataset via TF. Same for `TrajectoryDataset()`